From: Bruce Rogers <brogers@suse.com
Subject: acpiphp: add dmi info to acpiphp module
Patch-mainline: no
References: bnc#754391

To facilitate dynamically attaching and detaching devices
to a SLE11 KVM guest, we would like acpiphp automatically
loaded in that environment.  Add dmi information to the
acpiphp driver to detect a few of the "Bochs" strings used
in dmi data of the qemu / qemu-kvm bios.

Signed-off-by: Bruce Rogers <brogers@suse.com>

---
 drivers/pci/hotplug/acpiphp_core.c |   13 +++++++++++++
 1 file changed, 13 insertions(+)

Index: b/drivers/pci/hotplug/acpiphp_core.c
===================================================================
--- a/drivers/pci/hotplug/acpiphp_core.c
+++ b/drivers/pci/hotplug/acpiphp_core.c
@@ -38,6 +38,7 @@
 #include <linux/kernel.h>
 #include <linux/pci.h>
 #include <linux/pci_hotplug.h>
+#include <linux/dmi.h>
 #include <linux/slab.h>
 #include <linux/smp.h>
 #include "acpiphp.h"
@@ -371,6 +372,18 @@ void acpiphp_unregister_hotplug_slot(str
 		err("pci_hp_deregister failed with error %d\n", retval);
 }
 
+static struct dmi_system_id __initdata bochs_dmi_table[] = {
+	{
+		.ident = "Bochs",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Bochs"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Bochs"),
+		},
+	},
+	{}
+};
+
+MODULE_DEVICE_TABLE(dmi, bochs_dmi_table);
 
 static int __init acpiphp_init(void)
 {
